diff --git a/virttest/virsh.py b/virttest/virsh.py
index 9a2dca9..df0a898 100644
--- a/virttest/virsh.py
+++ b/virttest/virsh.py
@@ -511,6 +511,7 @@ def command(cmd, **dargs):
         cmd = "%s%s%s" % (virsh_exec, uri_arg, cmd)
         # Raise exception if ignore_status is False
         ret = utils.run(cmd, verbose=debug, ignore_status=ignore_status)
+        logging.debug(ret)
         # Mark return as not coming from persistent virsh session
         ret.from_session_id = None
 
@@ -1096,6 +1097,18 @@ def define(xml_path, **dargs):
     logging.debug("Define VM from %s", xml_path)
     return command(cmd, **dargs)
 
+def create(xml_path, **dargs):
+    """
+    Return True on successful domain create.
+
+    :param xml_path: XML file path
+    :param dargs: standardized virsh function API keywords
+    :return: CmdResult object
+    """
+    cmd = "create --file %s" % xml_path
+    logging.debug("Create VM from %s", xml_path)
+    return command(cmd, **dargs)
+
 
 def undefine(name, options=None, **dargs):
     """
