diff --git a/shared/cfg/guest-os/Linux/JeOS/19.x86_64.cfg b/shared/cfg/guest-os/Linux/JeOS/19.x86_64.cfg
index baede92..694e5ac 100644
--- a/shared/cfg/guest-os/Linux/JeOS/19.x86_64.cfg
+++ b/shared/cfg/guest-os/Linux/JeOS/19.x86_64.cfg
@@ -1,6 +1,6 @@
 - 19.x86_64:
     image_name = images/jeos-19-64
-    os_variant = fedora19
+    os_variant = fedora18
     vm_arch_name = x86_64
     shell_prompt = "^\[.*\][\#\$]\s*$"
     no unattended_install..floppy_ks
diff --git a/virttest/virsh.py b/virttest/virsh.py
index 3e59f38..df43dc2 100644
--- a/virttest/virsh.py
+++ b/virttest/virsh.py
@@ -612,7 +612,7 @@ def command(cmd, **dargs):
 
     virsh_exec = dargs.get('virsh_exec', VIRSH_EXEC)
     uri = dargs.get('uri', None)
-    debug = dargs.get('debug', False)
+    debug = dargs.get('debug', True)
     # Caller deals with errors
     ignore_status = dargs.get('ignore_status', True)
     session_id = dargs.get('session_id', None)
